baselines/power_of_choice/README.md

```diff
@@ -0,0 +1,250 @@
+```bash
+# cd to your power_of_choice directory (i.e. where the `pyproject.toml` is)
+
+# install the base Poetry environment
+poetry install
+
+# activate the environment
+poetry shell
+```
+
+## Running the Experiments
+
+First ensure you have activated your Poetry environment (execute `poetry shell` from this directory).
+
+### Generate Clients' Dataset
+First (and just the first time), the data partitions of clients must be generated.
+
+To generate the partitions for the FMNIST dataset (used in Figure 4 of the paper), run the following command:
+
+```bash
+# this will generate the datasets using the default settings in the `conf/base.yaml`
+python -m power_of_choice.dataset_preparation
+```
+
+The generated datasets will be saved in the `fmnist` folder.
+
+If you want to modify the `alpha` parameter used to create the LDA partitions, you can override the parameter:
+
+```bash
+python -m power_of_choice.dataset_preparation alpha=<alpha>
+```
+
+To generate partitions of the CIFAR10 dataset (used in Figure 6 of the paper), you can override the parameter:
+
+```bash
+python -m power_of_choice.dataset_preparation dataset.dataset="cifar10"
+```
+
+In this case the generated datasets will be saved in the `cifar10` folder.
+
+### Running simulations and reproducing results
+If you have not done it yet, [generate the clients' dataset](#generate-clients-dataset).
+
+
+#### MLP on FMNIST 
+
+The default configuration for `power_of_choice.main` uses the base version Power of Choice strategy with MLP on FMNIST dataset. It can be run with the following:
+
+```bash
+python -m power_of_choice.main # this will run using the default settings in the `conf/config.yaml`
+```
+
+You can override settings directly from the command line in this way:
+
+```bash
+python -m power_of_choice.main num_rounds=100 # will set the number of rounds to 100
+```
+
+To run using FedAvg:
+```bash
+# This will use FedAvg as strategy
+python -m power_of_choice.main variant="rand" 
+```
+
+To run all the experiments in Figure 4 of the paper, use the following commands:
+```bash
+# This will use FedAvg as strategy
+python -m power_of_choice.main variant="rand" 
+
+# This will use base version of Power of Choice with d=6
+python -m power_of_choice.main strategy.d=6
+
+# This will use base version of Power of Choice with d=9
+python -m power_of_choice.main strategy.d=9
+
+# This will use base version of Power of Choice with d=12
+python -m power_of_choice.main strategy.d=12
+```
+
+To run all the experiments in Figure 6 of the paper, use the following commands:
+```bash
+# This will use FedAvg as strategy
+python -m power_of_choice.main variant="rand" 
+
+# This will use base version of Power of Choice with d=6
+python -m power_of_choice.main strategy.d=6
+
+# This will use base version of Power of Choice with d=9
+python -m power_of_choice.main strategy.d=9
+
+# This will use base version of Power of Choice with d=12
+python -m power_of_choice.main strategy.d=12
+```
```

Comment:
I'm thinking maybe this is not needed since you already describe how to obtain Fig4,6 below.